newrelic-agent/src/main/java/com/newrelic/agent/InitProblemClasses.java

```diff
@@ -82,6 +82,15 @@ public static final void loadInitialClasses() {
             } catch (ClassNotFoundException e) {
                 Agent.LOG.log(Level.WARNING, "Error working around class loading issue:", e);
             }
+
+            // This works around a ClassCircularityError with ThreadLocalRandom.
```

Comment:
Can you give a little more context?
I think when the Error occurs and that it is related to Caffeine suffices.

This would help anyone reading the code a couple of years from now to decide whether this is still needed (without having to search for the ticket).